.appveyor.yml

```diff
@@ -0,0 +1,198 @@
+    driver: pdo_sqlsrv
+    db_version: sql2014
+    php: 7.1
+  - db: mssql
+    driver: sqlsrv
+    db_version: sql2016
+    php: 7.0
+  - db: mssql
+    driver: sqlsrv
+    db_version: sql2016
+    php: 7.1
+  - db: mssql
+    driver: pdo_sqlsrv
+    db_version: sql2016
+    php: 7.0
+  - db: mssql
+    driver: pdo_sqlsrv
+    db_version: sql2016
+    php: 7.1
+  - db: mysql
+    driver: pdo_mysql
+    php: 7.0
+  - db: mysql
+    driver: pdo_mysql
+    php: 7.1
+  - db: mysql
+    driver: mysqli
+    php: 7.0
+  - db: mysql
+    driver: mysqli
+    php: 7.1
+  - db: mariadb
+    driver: pdo_mysql
+    php: 7.0
+  - db: mariadb
+    driver: pdo_mysql
+    php: 7.1
+  - db: postgresql
+    driver: pdo_pgsql
+    php: 7.0
+  - db: postgresql
+    driver: pdo_pgsql
+    php: 7.1
+
+init:
+  - SET PATH=C:\Program Files\OpenSSL;c:\tools\php;%PATH%
+  - SET COMPOSER_NO_INTERACTION=1
+  - SET ANSICON=121x90 (121x90)
+
+## Install PHP and composer, and run the appropriate composer command
+install:
+    - ps: |
+        # Check if installation is cached
+        if (!(Test-Path c:\tools\php)) {
+          appveyor-retry cinst --ignore-checksums -y php --version ((choco search php --exact --all-versions -r | select-string -pattern $Env:php | Select-Object -First 1) -replace '[php|]','')
+          # install sqlite
+          appveyor-retry cinst -y sqlite
+
+          cd c:\tools\php
+
+          # Set PHP environment items that are always needed        
+          copy php.ini-production php.ini
+          Add-Content php.ini "`n date.timezone=UTC"
+          Add-Content php.ini "`n extension_dir=ext"
+          Add-Content php.ini "`n extension=php_openssl.dll"
+          Add-Content php.ini "`n extension=php_mbstring.dll"
+          Add-Content php.ini "`n extension=php_fileinfo.dll"
+          Add-Content php.ini "`n extension=php_pdo_sqlite.dll"
+          Add-Content php.ini "`n extension=php_sqlite3.dll"
+
+          # If needed get the MSSQL DLL's
+          if ($env:db -eq "mssql") {
+            $DLLVersion = "4.1.6.1"
+            cd c:\tools\php\ext
+            appveyor-retry appveyor DownloadFile http://windows.php.net/downloads/pecl/releases/sqlsrv/$($:DLLVersion)/php_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc14-x64.zip
+            7z x -y php_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc14-x64.zip > $null
+            appveyor-retry appveyor DownloadFile http://windows.php.net/downloads/pecl/releases/pdo_sqlsrv/$($DLLVersion)/php_pdo_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc14-x64.zip
+            7z x -y php_pdo_sqlsrv-$($DLLVersion)-$($env:php)-nts-vc14-x64.zip > $null
+            Remove-Item c:\tools\php\* -include .zip
+            cd c:\tools\php
+            Add-Content php.ini "`nextension=php_sqlsrv.dll"
+            Add-Content php.ini "`nextension=php_pdo_sqlsrv.dll"
+            Add-Content php.ini "`n"
+          }
+
+          # If needed set the the MySQL DLL's
+          if ($env:db -eq "mysql" -Or $env:db -eq "mariadb" ) {
```

Comment:
I thought you had made a request for the extensions that are packaged with PHP to be conditionally loaded, just like the services... must have been my misunderstanding. I'll remove those additional conditions.